### ISSUE TITLE
Support client-rendered pages

### ISSUE THREAD

**Issue by rosszurowski** [s3ththompson/new-work#3]
It'd be cool if `new-work` supported sites rendered on the client-side. I tried to add [Dan Holdsworth's](http://www.danholdsworth.com/) site to my list, but since it's rendered on the client with Angular, `newwork` can't detect any changes.

Some ideas on ways to handle this:
* Execute the client scripts before fetching the URL (I imagine this could get pretty messy)
* Allow defining a "source" URL that's different from the site URL. [His site uses a JSON API](http://www.danholdsworth.com/api/get/nodes/general?assets=query_thumbs_grouped,group,first&data=title,date,date2&device=desktop&filters=%7B%22node_type%22:%5B%22project%22,%22project_folder%22%5D,%22node_parent_id%22:3%7D&logged_in=0) to fetch the content (this won't work for every client-rendered site)

**Comment by s3ththompson** [comment]
Yep, this is definitely on the list. As you say, the easiest way to do this would be to allow an alternate source URL (overlap with #2 actually). Alternatively, I've been trying to wait for [headless Chrome](https://developers.google.com/web/updates/2017/04/headless-chrome) libraries to mature a bit so that we can scrape HTML after waiting for the page to render clientside (however, this will slow down scraping significantly).
